common/videoprofile_ids.go

```diff
@@ -13,3 +20,19 @@ var VideoProfileNameLookup = map[string]string{
 	"d435c53a": "P240p30fps4x3",
 	"fca40bf9": "P144p30fps16x9",
 }
+
+func videoProfileToBytes() VideoProfileByteMap {
+	var ret = make(VideoProfileByteMap, 0)
+	for k, v := range VideoProfileNameLookup {
+		b, err := hex.DecodeString(k)
+		if err != nil || len(b) != VideoProfileIDBytes {
+			continue
+		}
+		var key [VideoProfileIDBytes]byte
+		copy(key[:], b)
+		ret[key] = v
+	}
+	return ret
+}
+
+var VideoProfileByteLookup = videoProfileToBytes()
```

Comment:
Creates a lookup table translating each profile to its four-byte equivalents, rather than its hex-string equivalent as given by `VideoProfileLookup`. Used in `BytesToVideoProfile`.

This way we can look up video profiles based on its bytes without doing the int-to-hexstring conversion.

Reply:
Oh I see.  That explains my confusion with `VideoProfileToBytes` earlier.